File: bumperbot_cpp_examples/src/simple_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>
#include <chrono>

using namespace std::chrono_literals;

class SimplePublisher: public rclcpp::Node
{
    public:
        SimplePublisher(): Node("simple_publisher"), counter_(0)
        {
            pub_ = create_publisher<std_msgs::msg::String>("chatter", 10);
            timer_ = create_wall_timer(1s, std::bind(&SimplePublisher::TimerCallback, this));

            RCLCPP_INFO(get_logger(), "publishing at 1 HZ");
        }

    private:
        unsigned int counter_;
        rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub_;
        rclcpp::TimerBase::SharedPtr timer_;

        void TimerCallback()
        {
            std_msgs::msg::String msg = std_msgs::msg::String();
            msg.data = std::to_string(counter_);
            pub_->publish(msg);
            counter_ += 1;
        }
};


int main(int argc, char* argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimplePublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: bumperbot_python_examples/bumperbot_python_examples/simple_publisher.py
import rclpy
from rclpy.node import Node 
from std_msgs.msg import String

class SimplePublisher(Node):
    def __init__(self):
        super().__init__("simple_publisher")
        self.pub_ = self.create_publisher(String, "chatter", 10)
        self.counter_ = 0
        self.frequency = 1.0

        self.get_logger().info(f"Publishing at {self.frequency} HZ")
        self.timer_ = self.create_timer(1/self.frequency, self.timerCallback)

    def timerCallback(self):
        self.counter_ += 1
        msg = String()
        msg.data = f"timer: {self.counter_}"
        self.pub_.publish(msg)
        


def main(args=None):
    rclpy.init(args=args)
    node = SimplePublisher()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == "__main__":
    main()
    